gtests/net/mptcp/add_addr/add_addr_port_client.pkt

```diff
@@ -0,0 +1,25 @@
+// connection initiated by the kernel
+--tolerance_usecs=100000
+`../common/defaults.sh`
+
++0     `../common/client.sh`
+
++0.0   socket(..., SOCK_STREAM, IPPROTO_MPTCP) = 3
++0.0   setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
++0.0   fcntl(3, F_GETFL) = 0x2 (flags O_RDWR)
++0.0   fcntl(3, F_SETFL, O_RDWR|O_NONBLOCK) = 0
+
++0.0   connect(3, ..., ...) = -1 EINPROGRESS (Operation now in progress)
++0.0      >                             S   0:0(0)                    <mss 1460, sackOK, TS val 100 ecr 0,   nop, wscale 8, mpcapable v1 flags[flag_h] nokey>
++0.0      <                             S.  0:0(0)  ack 1  win 65535  <mss 1460, sackOK, TS val 700 ecr 100, nop, wscale 8, mpcapable v1 flags[flag_h] key[skey=2]>
++0.0      >                              .  1:1(0)  ack 1             <nop, nop,         TS val 100 ecr 700,                mpcapable v1 flags[flag_h] key[ckey, skey]>
+
++0.200  getsockopt(3, SOL_SOCKET, SO_ERROR, [0], [4]) = 0
++0.205  fcntl(3, F_SETFL, O_RDWR) = 0   // set back to blocking
+
+// ADD_ADDRESS with port
++0.0      <                              .  1:1(0)  ack 1 win 256                                                          <add_address address_id=12 addr[saddr] port=4321 hmac=auto, dss dack4=1 dll=0 nocs, nop, nop>
++0.0      >                              .  1:1(0)  ack 1             <nop, nop,          TS val 100 ecr 700,               add_address address_id=12 addr[saddr] port=4321 addr_echo, nop, nop>
+
+// TODO validate destination port in the outbound mp_join syn
```

Comment:
> is it in TODO because something is missing from the kernel side or packetdrill side?

I don't know it yet. The `SYN` packet in the commented line is matched only when packetdrill is launched without the wrapper, so I suspect that address parsing / packetdrill core networking might have some bug I'm still not aware of. But there might be something to check also in the kernel:  I'm unsure if  `address_id=0` in that MP_JOIN syn is compliant to [RFC8684](https://tools.ietf.org/html/rfc8684#section-3.4.1). See below: 
```
   Every address has an Address ID that can be used for uniquely
   identifying the address within a connection for address removal.  The
   Address ID is also used to identify MP_JOIN options (see Section 3.2)
   relating to the same address, even when address translators are in
   use.  The Address ID MUST uniquely identify the address for the
   sender of the option (within the scope of the connection); the
   mechanism for allocating such IDs is implementation specific.

   All Address IDs learned via either MP_JOIN or ADD_ADDR SHOULD be
   stored by the receiver in a data structure that gathers all the
   Address-ID-to-address mappings for a connection (identified by a
   token pair).
```

Reply:
For the `address_id=0`, that looks OK to me: is it not the sender -- the client here -- indicating the (source) ID address it is using?

So here, the sender is using the same address as the initial connection (ID=0) to establish a new subflow to a new address (ID=12 but represented by a new IP `saddr` (=`saddr0`?)). No?

Reply:
> For the `address_id=0`, that looks OK to me: is it not the sender -- the client here -- indicating the (source) ID address it is using?
> 
> So here, the sender is using the same address as the initial connection (ID=0) to establish a new subflow to a new address (ID=12 but represented by a new IP `saddr` (=`saddr0`?)). No?

ok, but how can the server "undoes" that `ADD_ADDR` ? by sending a `DEL_ADDR` having `id=0?`

Reply:
sorry, I'm not sure to understand the question. If an announced address -- via an `ADD_ADDR` or available in `MP_JOIN` (+ id 0 in the MP_CAP) -- is no longer available, the host should send an `RM_ADDR` for the corresponding ID. The ID can be 0 or any other number.

Reply:
> sorry, I'm not sure to understand the question. If an announced address -- via an `ADD_ADDR` or available in `MP_JOIN` (+ id 0 in the MP_CAP) -- is no longer available, the host should send an `RM_ADDR` for the corresponding ID. The ID can be 0 or any other number.

thanks for the explanation, if I well undrstand it the RFC says that the address id has a meaning for the *sender*. So, address id in that MP_JOIN is legal.